technico-v16/app/components/modal.tsx

```diff
@@ -0,0 +1,90 @@
+import React, { useState } from 'react';
+import Link from 'next/link';
+import { URLS } from '../lib/constants';
+
+type UserResponse = {
+  id: string;
+  name?: string;
+  surname?: string;
+};
+
+const ModalComponent: React.FC = () => {
+  const [users, setUsers] = useState<UserResponse[]>([]);
+  const [selectedUserId, setSelectedUserId] = useState<string | null>(null);
+
+  const openModal = async () => {
+    const modal = document.getElementById('my_modal_1') as HTMLDialogElement;
+    if (modal) {
+      modal.showModal();
+      await fetchUsers();
+    }
+  };
+
+  const fetchUsers = async () => {
```

Comment:
-   Good error handling in the fetch request with the try catch block
-   Consider adding loading state and better error messaging for users